plugins/avro-format/core/src/main/java/org/pentaho/di/trans/steps/avro/output/PentahoAvroOutputFormat.java

```diff
@@ -242,4 +251,67 @@ public void setVariableSpace( VariableSpace variableSpace ) {
     this.variableSpace = variableSpace;
   }
 
+  private Object mapDefaultValuesToDataTypes( String defaultValue, AvroSpec.DataType type ) {
+    try {
+      switch ( type ) {
+        case BOOLEAN:
+          return Boolean.parseBoolean( defaultValue );
+        case DATE:
+          return getDateTypeForDefaultValue( defaultValue );
+        case FLOAT:
+          return Float.parseFloat( defaultValue );
+        case DOUBLE:
+          return Double.parseDouble( defaultValue );
+        case LONG:
+        case INTEGER:
+          return Long.parseLong( defaultValue );
+        case DECIMAL:
+          return getDecimalTypeForDefaultValue( defaultValue );
+        case STRING:
+          return String.valueOf( defaultValue );
+        case BYTES:
+          return defaultValue.getBytes();
+        case TIMESTAMP_MILLIS:
+          return getTimeStampTypeForDefaultValue( defaultValue );
+        default:
+          return defaultValue;
+      }
+    } catch ( NumberFormatException | ParseException exception ) {
+      throw new IllegalArgumentException( "Not a valid default value " + defaultValue + " for data type " + type + " - " + exception.getMessage());
```

Comment:
Code change not necessary,

Do you need to include `exception.getMessage()` in this exception IllegalArgumentException message, The stack trace will have it's message below it.

Reply:
Added the message for user to track the custom exception message from the console.